src/reactor/mod.rs

```diff
@@ -530,7 +552,7 @@ impl Inner {
             Direction::Write => (&sched.writer, mio::Ready::writable()),
         };
 
-        task.register();
+        task.register_task(t);
```

Comment:
`task` is an `AtomicTask`? Not a big deal, just reads a little funny.

Reply:
I agree, but I also didn't want to have to re-run CI :P I'll try to bundle this up w/ another PR.